### PR TITLE
Fix for kNN distance calculation

### PR DESCRIPTION
**Pull request by KalyanovD**
# Description

### CODE REVIEW

File: cpp/daal/src/algorithms/service_kernel_math.h [hunk 1]
```diff
@@ -378,6 +403,12 @@ class ChebyshevDistances : public PairwiseDistances<FPType, cpu>
         return services::Status();
     }
 
+    virtual services::Status getRootBatch(const size_t n, FPType * in, FPType * out)
+    {
+        out = in;
```
Comment (SmirnovEgorRu) [inline cpp/daal/src/algorithms/service_kernel_math.h:408]: It does nothing outside the function

File: cpp/daal/src/algorithms/service_kernel_math.h [hunk 2]
```diff
@@ -171,6 +173,14 @@ class EuclideanDistances : public PairwiseDistances<FPType, cpu>
         return services::Status();
     }
 
+    virtual services::Status getRootBatch(const size_t n, FPType * in, FPType * out)
```
Comment (SmirnovEgorRu) [inline cpp/daal/src/algorithms/service_kernel_math.h:176]: better to align naming across distances
we have init(), we can call this as finalize() or so on

File: cpp/daal/src/algorithms/service_kernel_math.h [hunk 3]
```diff
@@ -294,6 +304,21 @@ class MinkowskiDistances : public PairwiseDistances<FPType, cpu>
...
+    {
+        daal::internal::mkl::MklMath<FPType, cpu> math;
+
+        if (_p == 1.0)
+        {
+            out = in;
```
Comment (SmirnovEgorRu) [inline cpp/daal/src/algorithms/service_kernel_math.h:313]: The same

**Review by SmirnovEgorRu — APPROVED** [review]

File: cpp/daal/src/algorithms/service_kernel_math.h (hunk 1 above)
Comment (SmirnovEgorRu) [reply cpp/daal/src/algorithms/service_kernel_math.h:408]: It's even a bug if ```in``` and ```out``` are different arrays.

File: cpp/daal/src/algorithms/service_kernel_math.h (hunk 2 above)
Comment (KalyanovD) [reply cpp/daal/src/algorithms/service_kernel_math.h]: applied, thanks for advise

File: cpp/daal/src/algorithms/service_kernel_math.h (hunk 1 above)
Comment (KalyanovD) [reply cpp/daal/src/algorithms/service_kernel_math.h]: set of arguments was changed as far as it's not necessary to have in and out arrays separate.

**Comment by KalyanovD** [comment]
perf is ok: http://intel-ci.intel.com/ebc9ddbb-1172-f12e-9f24-8cdcd4b723a1

**Review by PetrovKP — APPROVED** [review]

**Pull request merged**